File: solutions/283.move-zeroes/move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
function moveZeroes(nums) {
  let j = 0
  let len = nums.length

  for (let i = 0; i < len; i++) {
    let item = nums[i]
    if (item !== 0) {
      nums[j] = item
      j++
    }
  }

  while (j < len) {
    nums[j] = 0
    j++
  }

}

File: solutions/283.move-zeroes/move-zeroes.go
func moveZeroes(nums []int)  {
  	j := 0

	for i,val := range nums {
		if val != 0{
			nums[j] = nums[i]
			j++
		}
	}

	for j<len(nums) {
		nums[j] = 0
        j++
	}
 
}